src/Hedgehog/Shrink.fs

```diff
@@ -50,7 +48,7 @@ module Shrink =
     /// Shrink a list by edging towards the empty list.
     /// Note we always try the empty list first, as that is the optimal shrink.
     let list (xs : List<'a>) : seq<List<'a>> =
-        Seq.concat <| Seq.map (fun k -> removes k xs) (halves <| List.length xs)
+        Seq.collect (fun k -> removes k xs) (halves <| List.length xs)
```

Comment:
Good catch! :beers: (...wish there was an emoji with *3* beers! Fixed! :beers: + :beer:, so @TysonMN can join.)

Reply:
Ah, the ol' `map f >> flaten = bind f`.

Reply:
This was suggested to me by `ionide`, so I think the kudos should go to the analyzer :wink:

I don't know if this is more efficient, but it's definitely simpler.

Reply:
When I implement my own monads, sometimes I find it easier to directly implement `flatten` and then implement `bind f` as `map f >> flatten` and other times I find it easier to implement `bind` and then implement `flatten` as `bind id`.  Typically though I would expect `bind f` to be more efficient than `map f >> flatten`.